sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -861,17 +863,19 @@ lsc_scatter(AccessorTy acc, __ESIMD_NS::simd<uint32_t, N> offsets,
 /// @param acc is the SYCL accessor.
 /// @param offset is the zero-based offset in bytes.
 /// @param vals is values to store.
+/// @param pred is predicate.
 ///
 template <typename T, uint8_t NElts = 1,
           lsc_data_size DS = lsc_data_size::default_size,
           cache_hint L1H = cache_hint::none, cache_hint L3H = cache_hint::none,
           typename AccessorTy>
 __ESIMD_API std::enable_if_t<!std::is_pointer<AccessorTy>::value>
 lsc_block_store(AccessorTy acc, uint32_t offset,
-                __ESIMD_NS::simd<T, NElts> vals) {
+                __ESIMD_NS::simd<T, NElts> vals,
+                __ESIMD_NS::simd_mask<1> pred = 1) {
 #ifdef __ESIMD_FORCE_STATELESS_MEM
   lsc_block_store<T, NElts, DS, L1H>(
-      __ESIMD_DNS::accessorToPointer<T>(acc, offset), vals);
+      __ESIMD_DNS::accessorToPointer<T>(acc, offset), vals, pred);
 #else
   detail::check_lsc_vector_size<NElts>();
```

Comment:
Please make sure new functionality also works on ESIMD emulator, as `lsc` accesses work on emulator. @dongkyunahn-intel can help with logistics.

Reply:
I locally verified if two samples from https://github.com/intel/llvm-test-suite/pull/1194 are working fine with this PR. While `lsc_predicate.cpp` passed, `lsc_predicate_stateless.cpp` failed during compilation.

Building dpcpp toolchain supporting ESIMD_EMULATOR and running ESIMD kernels on emulator requires some environment setup. Let me know if you need help in locally verifying your implementation with ESIMD_EMULATOR.

Reply:
After `__ESIMD_FORCE_STATELESS_MEM` fix, both tests are passing.

```
$ SYCL/ESIMD/lsc/Output/lsc_predicate.cpp.tmp.out
Device name: ESIMD_EMULATOR
USM lsc predicate test passed
USM lsc predicate test passed
USM lsc predicate test passed
Accessor lsc predicate test passed
Accessor lsc predicate test passed
Accessor lsc predicate test passed
$ SYCL/ESIMD/lsc/Output/lsc_predicate_stateless.cpp.tmp.out
Device name: ESIMD_EMULATOR
Accessor lsc predicate test passed
Accessor lsc predicate test passed
Accessor lsc predicate test passed
```